session_3/blogspot/app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,81 @@
+class ArticlesController < ApplicationController
+  load_and_authorize_resource 
+  before_action :set_article, only: %w[ show edit update destroy ]
+
+  # GET /articles
+  # GET /articles.json
+  def index
+    @articles = Article.all
+  end
+
+  # GET /articles/1
+  # GET /articles/1.json
+  def show
+    if (not @article.public) && (@article.user_id != session[:user_id])
+      user = User.find_by(id: session[:user_id])
```

Comment:
This will be same as `current_user` variable present already, so you don't need to query again